frontend-ssr/utils/validation/userValidation.js

```diff
@@ -0,0 +1,28 @@
+const Joi= require("@hapi/joi");
```

Comment:
Never heard/used `@hapi/joi` before, but it seems the package has moved: https://github.com/sideway/joi/issues/2411

Also, is this they best way to validate? Doesn't mongoose offer something similar?

Reply:
I will look into it. Eventually returning with a mongoose version or own light `form` validation.

Reply:
Sooo, mongoose does have a form of validation with the schema as a did in the model with 
`{
name:{
type:String,
required:true
}}`

And you can do something like this with a small twist on the error handling 
`{
name:{
type:String,
required:[true,"Name is required"] 
}}`
This will create a custom error message inside the mongoose validation, if there is a custom message for every required field, we can collect all the errors inside the catch like 
` let save_errors={};
        for(let i in err.errors){
          save_errors[i]=err.errors[i].message;
        }`

The result looking like this 
`{
    "name": "Name is required",
    "username": "Username is required",
    "email": "Email is required"
}`